odoo/sql_db.py

```diff
@@ -114,24 +173,12 @@ def __init__(self):
     @check
     def savepoint(self, flush=True):
         """context manager entering in a new savepoint"""
-        name = uuid.uuid1().hex
+        cm = Savepoint(self)
         if flush:
-            flush_env(self, clear=False)
-            self.precommit.run()
-        self.execute('SAVEPOINT "%s"' % name)
-        try:
-            yield
-            if flush:
-                flush_env(self, clear=False)
-                self.precommit.run()
-        except Exception:
-            if flush:
-                clear_env(self)
-                self.precommit.clear()
-            self.execute('ROLLBACK TO SAVEPOINT "%s"' % name)
-            raise
-        else:
-            self.execute('RELEASE SAVEPOINT "%s"' % name)
+            cm = flushing(self, cm)
```

Comment:
This looks quite obscure to me.
The existing code was not very elegant, but it was explicit and easy to understand.
But that's only my opinion...